docs/sdk/src/reference_docs/tokens_in_substrate.rs

```diff
@@ -0,0 +1,97 @@
+//! has a use case requiring that an asset from a different pallet such as [`pallet_assets`] is
+//! used for the deposit. Rather than tightly couple [`pallet_preimage`] to [`pallet_balances`] AND
+//! [`pallet_assets`], along with every other token-handling pallet a user could possibly specify,
+//! [`pallet_preimage`] does not specify a concrete pallet as a dependency but instead accepts any
+//! dependency which implements the [`frame_support::traits::tokens::currency::ReservableCurrency`]
+//! trait. This allows [`pallet_preimage`] to support any arbitrary pallet implementing this trait,
+//! without needing any knowledge of what those pallets may be or requiring changes to support new
+//! pallets which may be written in the future.
+//!
+//! ## Fungible Token Traits in Substrate
+//!
+//! The [`frame_support::traits::fungible`] crate contains the latest set of Substrate
+//! fungible token traits, and is recommended to use for all new logic requiring a fungible token.
+//! See the crate documentation for more info about these fungible traits.
+//!
+//! [`frame_support::traits::fungibles`] provides very similar functionality to
+//! [`frame_support::traits::fungible`], except it supports managing multiple tokens.
+//!
+//! You may notice the trait [`frame_support::traits::Currency`] with similar functionality is also
+//! used in the codebase, however this trait is deprecated and existing logic is in the process of
+//! being migrated to [`frame_support::traits::fungible`] ([tracking issue](https://github.com/paritytech/polkadot-sdk/issues/226)).
+//!
+//! ## Fungible Token Trait Implementations in Substrate
+//!
+//! [`pallet_balances`] implements [`frame_support::traits::fungible`], and is the most commonly
+//! used fungible implementation in Substrate. Most of the time, it's used for managing the native
+//! token of the blockchain network it's used in.
+//!
+//! [`pallet_assets`] implements [`frame_support::traits::fungibles`], and is another popular
+//! fungible token implementation. It supports the creation and management of multiple assets in a
+//! single crate, making it a good choice when a network requires more assets in
+//! addition to its native token.
+//!
+//! ## Non-Fungible Tokens in Substrate
+//!
+//! [`pallet_uniques`] is recommended to use for all NFT use cases in Substrate.
+//! See the crate documentation for more info about this pallet.
+//!
+//! [`pallet_nfts`] is deprecated and should not be used.
+//!
+//! # What Next?
+//!
+//! - If you are interested in implementing a single fungible token, continue reading the
+//!   [`frame_support::traits::fungible`] and [`pallet_balances`] docs.
+//! - If you are interested in implementing a set of fungible tokens, continue reading the
+//!   [`frame_support::traits::fungibles`] trait and [`pallet_assets`] docs.
+//! - If you are interested in implementing an NFT, continue reading the [`pallet_uniques`] docs.
```

Comment:
Def supportive of the fact that this file is small and concise. 

Main issue I see is a slightly better intro that gives a bit more context + integrating better with `pallet_coupling.rs`.